File: 1041/1041.rb
x, y = gets.chomp.split(' ').map(&:to_f)

if x.zero? && y.zero?
  q = 0
elsif x > 0 && y > 0
  q = 1
elsif x < 0 && y > 0
  q = 2
elsif x < 0 && y < 0
  q = 3
else
  q = 4
end

if x.zero? && q.nonzero?
  puts 'Eixo Y'
elsif y.zero? && q.nonzero?
  puts 'Eixo X'
else
  puts q.zero? ? 'Origem' : "Q#{q}"
end

File: 1041/1041.cpp
#include <iostream>
#include <stdio.h>
using namespace std;

int main() {

  float x,y;
  scanf("%f %f",&x, &y);
  char q;

  if ((x == 0) && (y == 0)) {
    q = '0';
  }
  else if ((x > 0) && (y > 0)) {
    q = '1';
  }
  else if ((x < 0) && (y > 0)) {
    q = '2';
  }
  else if ((x < 0) && (y < 0)) {
    q = '3';
  }
  else { q = '4'; }

  if ((q == '0') && (x == 0)) {
    cout << "Origem\n";
  }
  else if (x == 0) {
    cout << "Eixo Y\n";
  }
  else if (y == 0) {
    cout << "Eixo X\n";
  }
  else {
    cout << 'Q' << q << '\n';
  }

  return 0;
}
